mne_features/feature_extraction.py

```diff
@@ -66,13 +67,19 @@ def transform(self, X, y='deprecated'):
             details.
         """
         X_out = super(FeatureFunctionTransformer, self).transform(X, y)
+        _feature_func = _get_python_func(self.func)
+        _params = self.get_params()
+        if hasattr(_feature_func, 'get_feature_names'):
+            self.feature_names = _feature_func.get_feature_names(X, **_params)
```

Comment:
I don't like that calling transform affects the state of the object. Only a fit is allowed to do this. Can you see a way out? also any attribute that is data dependent should end with _

Reply:
I see. Then, what about
```
    def fit(self, X, y=None):
        """Fit the FeatureFunctionTransformer (does not extract features).
        
        Parameters
        ----------
        X : ndarray, shape (n_channels, n_times)
        
        y : ignored
        
        Returns
        -------
        self
        """
        self._check_input(X)
        _feature_func = _get_python_func(self.func)
        _params = self.get_params()
        if hasattr(_feature_func, 'get_feature_names'):
            self.feature_names = _feature_func.get_feature_names(X, **_params)
        return self
```
in `FeatureFunctionTransformer`?

Reply:
yes this is ok in terms of API but `_params = self.get_params()` could be in the if block